### PR TITLE
Add support for LDAP

### PR DESCRIPTION
**Pull request by syyyr**

### CODE REVIEW

File: libshvbroker/src/brokerapp.cpp [hunk 1]
```diff
@@ -517,9 +558,86 @@ bool BrokerApp::checkTunnelSecret(const std::string &s)
 	return m_tunnelSecretList.checkSecret(s);
 }
 
-chainpack::UserLoginResult BrokerApp::checkLogin(const chainpack::UserLoginContext &ctx)
+class AuthThread : public QThread {
```
Comment (fvacek) [inline libshvbroker/src/brokerapp.cpp:561]: `LdapAuthThread`

File: libshvbroker/src/brokerapp.cpp [hunk 2]
```diff
@@ -517,9 +558,86 @@ bool BrokerApp::checkTunnelSecret(const std::string &s)
...
+				emit resultReady(chainpack::UserLoginResult{false, err.what()});
+			}
+			return;
+		}
+#endif
+
+		emit resultReady(chainpack::UserLoginResult{false});
+	}
+
+	Q_SIGNAL void resultReady(const chainpack::UserLoginResult& s);
```
Comment (fvacek) [inline libshvbroker/src/brokerapp.cpp:611]: I still feel, that `Q_SIGNAL void resultReady(const chainpack::UserLoginResult& s, QStringList user_groups);` or extending `UserLoginResult` by field `userLoginGroups` will be better design. I do not like to use 2 signals for one event.

**Review by fvacek — CHANGES_REQUESTED** [review]

File: libshvbroker/src/brokerapp.cpp [hunk 3]
```diff
@@ -517,9 +558,86 @@ bool BrokerApp::checkTunnelSecret(const std::string &s)
...
+#endif
+		: m_ctx(ctx)
+#ifdef WITH_SHV_LDAP
+		, m_ldapConfig(ldap_config)
+#endif
+	{
+	}
+	void run() override
+	{
+		if (auto result = BrokerApp::instance()->aclManager()->checkPassword(m_ctx); result.passwordOk) {
```
Comment (fvacek) [inline libshvbroker/src/brokerapp.cpp:577]: still access to foreing thread object

Comment (syyyr) [reply libshvbroker/src/brokerapp.cpp]: fixed by not doing this in the thread

File: libshvbroker/src/brokerapp.cpp (hunk 1 above)
Comment (syyyr) [reply libshvbroker/src/brokerapp.cpp]: done

**Review by fvacek — APPROVED** [review]

**Pull request merged**